src/pycsou/opt/solver/pds.py

```diff
@@ -0,0 +1,965 @@
+    * The problem is *feasible* --i.e. there exists at least one solution.
+
+    **Remark 1:**
+
+    The algorithm is still valid if one of the terms :math:`\mathcal{F}` or :math:`\mathcal{G}` is zero.
+
+    **Remark 2:**
+
+    Assume that the following holds:
+      - :math:`\frac{1}{\tau}\geq \frac{\beta}{2}`,
+      - :math:`\rho \in ]0,\delta[`, where :math:`\delta:=2-\frac{\beta}{2}\tau\in[1,2[.`
+
+    Then, there exists a pair :math:`(\mathbf{x}^\star,\mathbf{z}^\star)\in\mathbb{R}^N\times \mathbb{R}^M`} solution s.t. the primal and dual sequences of  estimates :math:`(\mathbf{x}_n)_{n\in\mathbb{N}}` and :math:`(\mathbf{z}_n)_{n\in\mathbb{N}}` *converge* towards :math:`\mathbf{x}^\star` and :math:`\mathbf{z}^\star` respectively, i.e.
+
+    .. math::
+       \lim_{n\rightarrow +\infty}\Vert\mathbf{x}^\star-\mathbf{x}_n\Vert_2=0, \quad \text{and} \quad  \lim_{n\rightarrow +\infty}\Vert\mathbf{z}^\star-\mathbf{z}_n\Vert_2=0.
+
+    **Default values of the hyperparameters provided here always ensure convergence of the algorithm.**
+
+    ``ForwardBackwardSplitting.fit()`` **Parameterization**
+
+    x0: NDArray
+        (..., N) initial point(s) for the primal variable.
+    z0: NDArray
+        (..., N) initial point(s) for the dual variable.
+        If None (default), then use x0 as the initial point(s) for the dual variable as well.
+    tau: Real
+        Primal step size
+    rho: Real
+        Momentum parameter
+    beta: Real
+        Lipschitz constant :math:`\beta` of the derivative of :math:`\mathcal{F}`.
+
+
+
+    See Also
+    --------
+    :py:class:`~pycsou.opt.solver.pds.FBS`, :py:class:`~pycsou.opt.solver.pgd.PGD`
+    """
+
+    def __init__(
+        self,
+        f: typ.Optional[pyco.ProxFunc] = None,
+        g: typ.Optional[pyco.ProxFunc] = None,
+        *,
+        folder: typ.Optional[pyct.PathLike] = None,
+        exist_ok: bool = False,
+        writeback_rate: typ.Optional[int] = None,
+        verbosity: int = 1,
+        log_var: pyct.VarName = ("x",),
+    ):
+        super(ForwardBackwardSplitting).__init__(
+            f=f,
+            g=g,
+            h=None,
+            K=None,
+            folder=folder,
+            exist_ok=exist_ok,
+            writeback_rate=writeback_rate,
+            verbosity=verbosity,
+            log_var=log_var,
+        )
+
+    @pycrt.enforce_precision(i=["x0", "z0", "tau", "rho", "beta"], allow_None=True)
+    def m_init(
+        self,
+        x0: pyct.NDArray,
+        z0: typ.Optional[pyct.NDArray],
+        tau: typ.Optional[pyct.Real] = None,
+        rho: typ.Optional[pyct.Real] = 1.0,
+        beta: typ.Optional[pyct.Real] = None,
+    ):
+        mst = self._mstate  # shorthand
+        mst["x"] = mst["x_prev"] = x0
+        mst["z"] = mst["z_prev"] = self.set_dual_variable(z0)
+        mst["tau"] = tau
+        mst["beta"] = self.set_beta(beta)
+        mst["rho"] = rho
+
+
+FBS = ForwardBackwardSplitting
+
+
+class PD3O(pycs.Solver):
```

Comment:
This class is extremely redundant with PDS. We need a superclass. Also the step sizes are incorrectly set here since PD3O has different convergence properties.